src/pages/ResetPassword/ResetPassword.tsx

```diff
@@ -0,0 +1,120 @@
+import { useLocation, useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+import Modal from "../../components/modals/Modal";
+import Button from "../../components/Button";
+import NewPasswordInput from "../../components/NewPasswordInput";
+import { useMutation } from "react-query";
+import toast from "react-hot-toast";
+import ToastMessage from "../../components/ToastMessage";
+import { useApi } from "../../contexts/ApiProvider";
+import Spinner from "../../components/Spinner";
+import Row from "../../components/layout/Row";
+import TextField from "../../components/TextField";
+
+export default function ResetPassword() {
+  const { api } = useApi();
+  const navigate = useNavigate();
+  const resetToken = new URLSearchParams(useLocation().search).get("token");
+
+  const [password, setPassword] = useState("");
+  const [hasPasswordError, setHasPasswordError] = useState(true);
+  const [triedToSubmit, setTriedToSubmit] = useState(false);
+  const [recover, setRecover] = useState(false);
+  const [recoveryPhrase, setRecoveryPhrase] = useState("");
+
+  const [error, setError] = useState("");
+
+  useEffect(() => {
+    // redirect to home if no token is provided
+    if (resetToken) {
+      window.HSOverlay.open(document.querySelector("#reset-pass-modal"));
```

Comment:
I would wrap this in a try catch just to be safe, I've seen this function crash the code if it tries to load the modal before the page is loaded.

Reply:
added a check to make sure the modal plugin is loaded